CMakeLists.txt

```diff
@@ -55,7 +55,6 @@ xyz_add_library(
 target_sources(indirect_cxx14
     INTERFACE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/indirect_cxx14.h>
-    PRIVATE
```

Comment:
Do we want these files to be visible to users of this library? I would expect them to only care about the headers containing the implementation?

Reply:
If I don’t put them here then they don’t show up in the compilation commands JSON file. 

I don’t know of a better fix, this works but is not ideal.

Reply:
I'm surprised that they would have to be declared as interface files to create a compile command.  The `INTERFACE` keyword in effect says they sound not be compiled within the indirect_cxx17 project, but should become part of the depending project.  

Is that the issue that you compile command for this file in a downstream target?

Reply:
Oh I think I see, `xyz_add_library` creates an `INTERFACE` library so then does not create compile commands for it be cause the `INTERFACE` keyword explicitly says things from this target are needed for a downstream target only.

We could change the library to just be a static library (but this may cause linkage errors as there would be nothing to link) or move these to the test targets

Reply:
My CMake taste is nowhere near as refined as yours. What would you prefer? 

Whatever we do needs an explanatory comment as it's weird. Having an empty source file is weird anyway so weird is perhaps to be expected.

Reply:
Hmmm... what was the context for needing to compile these (that we did not get over doing so in the tests)?

If we don't really want it to be part of the library but want to compile then I would suggest were make a separate target per library which is an `OBJECT` library.  This is a library of compiled translation units.  Then we can keep them separate but still ensure we have a check which compiles each header (just they will become a separate target)

Does that make sense?  I can do this if that helps...

Reply:
This sounds like the right fix - go team!

Reply:
I've applied the fix @Twon suggested - worked easily.

Reply:
Great, glad that has worked!